config/provider-configuration.yaml

```diff
@@ -44,3 +44,11 @@ supported_providers:
         supported_instance_type:
           standard: {}
           developer: {}
+  - name: gcp
```

Comment:
I am thinking about what would be our preferred defaults regarding the provider configuration and dynamic scaling functionality including both AWS and GCP for our development environment.

With the proposed configuration (and also with already pre-existing configuration with AWS) when dynamic scaling is enabled 4 new clusters would be created: one for each supported instance type in each supported cloud provider. This could easily create undesired clusters when people are developing without realizing about it.

Maybe we should by default limit the scaling of clusters in the development configuration?
Two options I can think of if we were to decide that:
1. Set a limit of 0 for all instance types for all providers in the provider configuration
2. Set the `EnableDynamicScaleUpManagerScaleUpTrigger` and its corresponding scaledown option to false by default in development environment.

I prefer the first option as it would be more explicit and the.
In this case, if someone wants to really trigger scale ups would need to modify the limit for the desired instance type in the configuration file.

cc @machi1990 @pb82

Reply:
adding @JameelB @JasonPowr  @VanillaSpoon to the discussion as well

Reply:
I am thinking maybe we could keep the scaling mode to `manual` for development as the default.
And have section in the documentation explaining how to switch to `auto` mode: this will contain a  `NOTE` on the implication of having `auto` mode etc

Reply:
@machi1990 do you mean when `OCM_ENV` is set to development? We can do that, but people could still set the env to a different value when running locally.

Reply:
> @machi1990 do you mean when `OCM_ENV` is set to development? 

Yes. The scaling type is already set to `manual` as the default. https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/dc3eb0e33955fa8329b44dd58bdbc6bfe4599ef9/internal/kafka/internal/environments/development.go#L29

> We can do that, but people could still set the env to a different value when running locally.

For other environments, if scaling type is not specified manual will be the default scaling type. https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/dc3eb0e33955fa8329b44dd58bdbc6bfe4599ef9/internal/kafka/internal/config/dataplane_cluster_config.go#L90

Essentially, we are saying that by default we won't turn autoscaling on. When someone turns it on, they understand the impact (scaling up/down, taints addition etc)

Reply:
Ok, that sounds reasonable to me. Do we need documentation for turning on dynamic scaling for development and what the implications are?

Reply:
We already have docs here https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/docs/architecture/data-plane-osd-cluster-dynamic-scaling.md

Which is already linked from https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/docs/data-plane-osd-cluster-options.md#configuring-osd-cluster-creation-and-autoscaling 

We could add a big `NOTE` there that someone reads the architectural proposal to understand the implication of turning dynamic scaling on.